src/ParallelAlgorithms/Actions/SetData.hpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace Actions {
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Mutate the DataBox tags in `TagsList` according to the `data`.
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - All tags in `TagsList`
+ */
+template <typename TagsList>
+struct SetData;
+
+template <typename... Tags>
+struct SetData<tmpl::list<Tags...>> {
+  template <
+      typename ParallelComponent, typename DataBox, typename Metavariables,
+      typename ArrayIndex,
+      Requires<cpp17::conjunction_v<db::tag_is_retrievable<Tags, DataBox>...>> =
+          nullptr>
```

Comment:
So I can learn, what is the nullptr doing? I get that the overall effect is to require that all tags are retrievable, just wondering why you assign the Requires<> type to nullptr. Also wondering why Requires<> isn't taking a type (e.g. cpp17::conjunction_t)?

Reply:
Requires is our replacement for std::enable_if